### ISSUE TITLE
Keyvault Secret error message when creating 16 tags or more

### ISSUE THREAD

**Issue by scott-doyland-burrows** [hashicorp/terraform-provider-azurerm#16780]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.9

### AzureRM Provider Version

3.5.0

### Affected Resource(s)/Data Source(s)

azurerm_key_vault_secret

### Terraform Configuration Files

```hcl
data "azurerm_key_vault" "azurerm_key_vault" {
  name                = "example-kv"
  resource_group_name = "example-rg"
}

resource "azurerm_key_vault_secret" "secret" {
  name         = "example-secret-name"
  value        = "example-secret-value"
  key_vault_id = data.azurerm_key_vault.azurerm_key_vault.id

  tags = {
    "a" = "1"
    "b" = "2"
    "c" = "3"
    "d" = "4"
    "e" = "5"
    "f" = "6"
    "g" = "7"
    "h" = "8"
    "i" = "9"
    "j" = "10"
    "k" = "11"
    "l" = "12"
    "m" = "13"
    "n" = "14"
    "o" = "15"
    "p" = "16"
  }
}
```

### Debug Output/Panic Output

```shell
Error: keyvault.BaseClient#SetSecret: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadParameter" Message="Property  has invalid value\r\n"
│
│   with azurerm_key_vault_secret.secret,
│   on terraform.tf line 6, in resource "azurerm_key_vault_secret" "secret":
│    6: resource "azurerm_key_vault_secret" "secret" {
```

### Expected Behaviour

The error message should be clear that only 15 tags are allowed.
The ```terraform plan``` should indicate that only 15 tags are allowed (currently the ```plan``` works but the ```apply``` fails).

### Actual Behaviour

The error above occurs when there are 16 or more tags defined.

Azure only allows 15 tags - the error message should be clear, and should be shown in the ```plan``` as to what the failure is, as it took me about two hours to work it out.

### Steps to Reproduce

Setup a keyvault and create a secret with 16 tags or more.

**Comment by sinbai** [comment]
@scott-doyland-burrows since the issue has been released in [v3.9.0 of the Terraform Provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.9.0/CHANGELOG.md), are you okey if we close this issue?

**Comment by scott-doyland-burrows** [comment]
Hi,

I cannot see in the changelog that the resource has been fixed?

Regards,
Scott

**Comment by sinbai** [comment]
@scott-doyland-burrows thanks for noticing that it was indeed not included in the changelog. However, I confirmed that the fix has been released in the v3.9.0 of the Terraform Provider. Could you upgrade  Terraform Provider to see if the issue is fixed?

**Comment by mybayern1974** [comment]
Per sinbai's suggestion that the fix has been included in v3.9.0, I'm closing this issue. Feel free to call out if this issue needs to be re-activated.
